ledger/src/blockstore_processor.rs

```diff
@@ -493,7 +548,9 @@ fn process_entries_with_callback(
                 }
 
                 if randomize {
-                    transactions.shuffle(&mut rng);
+                    let indices = get_random_indices(transactions.len());
```

Comment:
Ugh this randomize thing has always been a PITA

I would prefer we stick with the generic `shuffle` implementation because that's probably more optimized than a one-off implementation we then maintain.

1. I think we can zip the transactions and the indices, via `let transactions_and_indices = transactions.iter().zip(start_index..).collect::Vec<(Transaction, usize)>()`
2. Shuffle that vector from 1, `transactions_and_indices.shuffle(&mut rng)`.
3. Unzip into indices and transaction_indexes `let (transactions, indices): (Vec<Transaction>, Vec<usize>) = transactions.into_iter().unzip()`

Reply:
Side note doing this, I don't think we need to pass `entry_indexes: &mut [Vec<usize>],` to this function, we just need the `entry_indexes: &mut [usize]`, i.e. just the starting index of each entry.

Reply:
Just for the record, this implementation is the same as what `shuffle` is doing under the hood. But okay, I'll zip/unzip

Reply:
This is making me sad. Haven't yet convinced the compiler to let me do that zip/unzip due to the handling of transactions in the following loop

Reply:
Finally got it! Needed to drain the original txs vec